kdTree.cpp

```diff
@@ -0,0 +1,344 @@
+#include "precomp.h"
+#include "kdTree.h"
+
+namespace Tmpl8 {
+	kdTree::kdTree()
+	{
+
+	}
+
+    /// <summary>
+    /// creates a new node
+    /// </summary>
+    /// <param name="tank"> a pointer to a tank </param>
+    /// <returns>returns a pointer to the just made node</returns>
+    kdTree::node* kdTree::newnode(Tank* tank)
+	{
+
+        struct node* node
+            = (struct node*)malloc(sizeof(struct node));
```

Comment:
Using malloc is bad practice because it returns a raw void pointer to uninitialized memory. That's why you have to cast it to a node pointer.

If you want to instance it on the stack just use a normal instantiation:
> When initializing a [struct](https://en.cppreference.com/w/c/language/struct), the first initializer in the list initializes the first declared member (unless a designator is specified)(since C99), and all subsequent initializers without designators (since C99)initialize the struct members declared after the one initialized by the previous expression.
```
struct point {double x,y,z;} p = {1.2, 1.3}; // p.x=1.2, p.y=1.3, p.z=0.0
div_t answer = {.quot = 2, .rem = -1 };      // order of elements in div_t may vary
```
[source](https://en.cppreference.com/w/c/language/struct_initialization)

to obtain a pointer to a struct you can simply use:
```
auto struct_ptr = &struct
```